src/routers/alpha-router/functions/get-candidate-pools.ts

```diff
@@ -1257,6 +1386,9 @@ export async function getMixedRouteCandidatePools({
     { subgraphPools: V2subgraphPools, candidatePools: V2candidatePools },
   ] = [v3CandidatePools, v2CandidatePools];
 
+  V2subgraphPools.push(...crossLiquidityPools.v2Pools);
+  V3subgraphPools.push(...crossLiquidityPools.v3Pools);
```

Comment:
Here we "inject" the cross liquidity pools to each protocol pools.

These are injected at the end before we calculate the mixed routes, but it works because of how the rest of the BL is defined.